pytorch_lightning/overrides/torch_distributed.py

```diff
@@ -0,0 +1,177 @@
+# type: ignore
+
+import io
+import logging
+import os
+import pickle
+
+import torch
+from torch._C._distributed_c10d import ProcessGroup
+
+from pytorch_lightning.utilities import _HPU_AVAILABLE
+from pytorch_lightning.utilities.imports import _TORCH_GREATER_EQUAL_1_8
+
+_pickler = pickle.Pickler
+_unpickler = pickle.Unpickler
+
+
+logger = logging.getLogger(__name__)
+
+if torch.distributed.is_available():
+    from torch.distributed import Backend, broadcast, get_backend, get_rank, GroupMember
+
+# The code underneath is taken from PyTorch `torch/distributed/distributed_c10d.py`
+# the distributed backend and tensor type updates for habana backend is done here before broadcast
+
+
+# https://github.com/pytorch/pytorch/blob/1.10.1/torch/distributed/distributed_c10d.py#L256
+def _rank_not_in_group(group: ProcessGroup):
+    """Helper that checks if the current process's rank is not in a given group."""
+    if group is None:
+        return False
+    return group == GroupMember.NON_GROUP_MEMBER
+
+
+# Taken from https://github.com/pytorch/pytorch/blob/1.10.1/torch/distributed/distributed_c10d.py#L1518
+def _object_to_tensor(obj):
+    f = io.BytesIO()
+    _pickler(f).dump(obj)
+    byte_storage = torch.ByteStorage.from_buffer(f.getvalue())
+    # Do not replace `torch.ByteTensor` or `torch.LongTensor` with torch.tensor and specifying dtype.
+    # Otherwise, it will casue 100X slowdown.
+    # See: https://github.com/pytorch/pytorch/issues/65696
+    byte_tensor = torch.ByteTensor(byte_storage)
+    local_size = torch.LongTensor([byte_tensor.numel()])
+    return byte_tensor, local_size
+
+
+# Taken from https://github.com/pytorch/pytorch/blob/1.10.1/torch/distributed/distributed_c10d.py#L1530
+def _tensor_to_object(tensor, tensor_size):
+    buf = tensor.numpy().tobytes()[:tensor_size]
+    return _unpickler(io.BytesIO(buf)).load()
+
+
+# Taken from https://github.com/pytorch/pytorch/blob/1.10.1/torch/distributed/distributed_c10d.py#L1729
+def _broadcast_object_list(object_list, src=0, group=None, device=None):
```

Comment:
```suggestion
def _broadcast_object_list(object_list, source_rank=0, group=None, device=None):
```
let's be more descriptive

Reply:
`src` is the standard variable within torch distributed